core/controllers/topic_editor_test.py

```diff
@@ -70,6 +77,71 @@ def test_story_creation(self):
         self.logout()
 
 
+class SubtopicPageEditorTest(BaseTopicEditorControllerTest):
+
+    def test_editable_subtopic_page_get(self):
+        # Check that non-admins cannot access the editable subtopic data.
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
+            self.login(self.NEW_USER_EMAIL)
+            response = self.testapp.get(
+                '%s/%s/%s' % (
+                    feconf.SUBTOPIC_PAGE_EDITOR_DATA_URL_PREFIX,
+                    self.topic_id, 1),
+                expect_errors=True)
+            self.assertEqual(response.status_int, 401)
+            self.logout()
+
+            # Check that admins can access the editable subtopic data.
+            self.login(self.ADMIN_EMAIL)
+            json_response = self.get_json(
+                '%s/%s/%s' % (
+                    feconf.SUBTOPIC_PAGE_EDITOR_DATA_URL_PREFIX,
+                    self.topic_id, 1))
+            subtopic_page_id = topic_domain.SubtopicPage.get_subtopic_page_id(
+                self.topic_id, 1)
+            self.assertEqual(
+                subtopic_page_id, json_response['subtopic_page']['id'])
+            self.logout()
+
+    def test_editable_subtopic_page_put(self):
+        # Check that admins can edit a topic.
```

Comment:
Check topic managers too.

In general please go through the tests for controllers and bulk them up a bit to ensure that there are correct tests for the various user categories (including e.g. topic managers trying to edit a subtopic that belongs to another topic manager). It's important to get this right.

Reply:
Done.